contracts/neutron-interchain-txs/src/contract.rs

```diff
@@ -0,0 +1,417 @@
+// Copyright 2022 Neutron
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use cosmos_sdk_proto::cosmos::base::v1beta1::Coin;
+use cosmos_sdk_proto::cosmos::staking::v1beta1::{
+    MsgDelegate, MsgUndelegate, MsgUndelegateResponse,
+};
+#[cfg(not(feature = "library"))]
+use cosmwasm_std::entry_point;
+use cosmwasm_std::{
+    from_binary, to_binary, to_vec, Binary, CosmosMsg, Deps, DepsMut, Env, MessageInfo, Reply,
+    Response, StdError, StdResult, Storage, SubMsg,
+};
+use interchain_txs::helpers::{parse_item, parse_response};
+use interchain_txs::msg::SudoMsg;
+use interchain_txs::storage::RequestPacket;
+use neutron_bindings::msg::NeutronMsg;
+use neutron_bindings::query::InterchainQueries;
+
+use neutron_bindings::types::QueryInterchainAccountAddressResponse;
+use neutron_bindings::ProtobufAny;
+use prost::Message;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use crate::error::ContractResult;
+use crate::msg::{InstantiateMsg, MigrateMsg};
+use crate::storage::{
+    Config, CONFIG, IBC_SUDO_ID_RANGE_END, IBC_SUDO_ID_RANGE_START, REPLY_QUEUE_ID, SUDO_PAYLOAD,
+};
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum QueryMsg {
+    Ica {},
+    Config {},
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+struct OpenAckVersion {
+    version: String,
+    controller_connection_id: String,
+    host_connection_id: String,
+    address: String,
+    encoding: String,
+    tx_type: String,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum ExecuteMsg {
+    Delegate { validator: String, amount: u128 },
+    Undelegate { validator: String, amount: u128 },
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub struct SudoPayload {
+    pub message: String,
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn instantiate(
+    deps: DepsMut,
+    _env: Env,
+    _info: MessageInfo,
+    msg: InstantiateMsg,
+) -> ContractResult<Response<NeutronMsg>> {
+    let register = NeutronMsg::register_interchain_account(
+        msg.connection_id.clone(),
+        msg.interchain_account_id.clone(),
+    );
+    CONFIG.save(
```

Comment:
I don't think that the information we save here is `CONFIG` information. Also I think that  we should not register the interchain account in `instantiate()`, there should be a separate handler for registering an interchain account — at lest because we'll need to test registering multiple accounts on the same remote zone